doc/builders/images/dockertools.section.md

```diff
@@ -394,3 +394,114 @@ buildImage {
   };
 }
 ```
+
+## buildNixShellImage {#ssec-pkgs-dockerTools-buildNixShellImage}
+
+Create a Docker image that sets up the `nix-shell` environment of a derivation. With Docker, this environment is similarly sandboxed as if the derivation were built with `nix-build`, with a major difference being that access to the internet is allowed. It additionally also behaves like an interactive `nix-shell`, running things like `shellHook` and setting an interactive prompt. If the derivation is fully buildable (i.e. `nix-build` can be used on it), running `buildDerivation` inside such a Docker image will build the derivation, with all its outputs being available in the correct `/nix/store` paths, pointed to by the respective environment variables like `$out`, etc.
+
+::: {.warning}
+The behavior doesn't match nix-shell exactly and this function may not work correctly for e.g. fixed-output derivations, CA derivations, impure derivations and other special types of derivations.
+:::
+
+`drv`
+
+: The derivation whose environment this Docker image should be based on.
+
+    Adding additional packages to the Docker image is possible by e.g. extending the list of `nativeBuildInputs` of this derivation like
+
+    ```nix
+    buildNixShellImage {
+      drv = someDrv.overrideAttrs (old: {
+        nativeBuildInputs = old.nativeBuildInputs or [] ++ [
+          somethingExtra
+        ];
+      });
+      # ...
+    }
+    ```
+
+    Similarly, you can extend the image initialization script by extending `shellHook`
+
+`name` _optional_
+
+: The name of the resulting image.
+
+    *Default:* the `drv`'s name + `"-env"`
+
+`tag` _optional_
+
+: Tag of the generated image.
+
+    *Default:* the resulting image derivation output path's hash
+
+`uid`/`gid` _optional_
+
+: The user/group id to run the container as. This is like a nixbld build user
+
+    *Default:* 1000/1000
+
+`homeDirectory` _optional_
+
+: The home directory of the user
+
+    *Default:* `/build`
+
+`shell` _optional_
+
+: The path to the bash binary to use as the shell. This shell is started when running the image
+
+    *Default:* `pkgs.bashInteractive + "/bin/bash"`
+
+`command` _optional_
+
+: Run this command in the environment of the derivation, in an interactive shell. See `--command` in `man nix-shell`
```

Comment:
Why don't we link to the Nix manual here?

```suggestion
: Run this command in the environment of the derivation, in an interactive shell. See `--command` in `man nix-shell`.
```

Reply:
Good idea, doing that